contracts/InvestProvider.sol

```diff
@@ -12,11 +12,14 @@ contract InvestProvider is InvestInternal {
     /// @dev Constructor to initialize the contract with a `lockDealNFT`.
     /// @param _lockDealNFT The address of the `ILockDealNFT` contract.
     /// @param _dispenserProvider The address of the `IProvider` contract for dispensers.
-    constructor(ILockDealNFT _lockDealNFT, IProvider _dispenserProvider) {
+    /// @param _wBNB The address of the `IWBNB` contract.
+    constructor(ILockDealNFT _lockDealNFT, IProvider _dispenserProvider, IWBNB _wBNB) {
         if (address(_lockDealNFT) == address(0)) revert NoZeroAddress();
         if (address(_dispenserProvider) == address(0)) revert NoZeroAddress();
+        if (address(_wBNB) == address(0)) revert NoZeroAddress();
         lockDealNFT = _lockDealNFT;
         dispenserProvider = _dispenserProvider;
+        wBNB = _wBNB;
```

Comment:
Don't need to save. let's use the address from the Pool ID

Reply:
Do you mean tokens should be marked something like this? 
```solidity
    function createNewPool(
        uint256 poolAmount,
        uint256 sourcePoolId,
        bool isWrapped
    )
```